source/userguide/kb.rst

```diff
@@ -813,3 +813,122 @@ We can perform the same search using the :meth:`query()` API as well.
+
+The 50, 100, 200, and 300 dimension word embeddings are provided. The desired dimension can be specified via the ``token_embedding_dimension`` in the ``model_settings`` section of the config. For fields with more than one token, word vector averaging is used. 
+
+Once your config file is specified, you can load the Knowledge Base and the embeddings will automatically be generated as specified in the config and stored in your index as dense vectors. Note that you must pass the application path in order for the config to be processed. This can be done via the CLI command with the ``--app-path`` option or via the ``app_path`` parameter for the ``load_kb`` method.
+
+
+.. code-block:: console
+
+  mindmeld load-kb hr_assistant faq_emb data/hr_faq_data.jsonl --app_path ./
+
+All of the vectors generated at load time will be cached for faster retrieval at inference time and for future loads. It is stored with other generated data in the generated folder under the provided model name. It’s important to update the mode name when updating the model settings to maintain consistency with the cache. 
+
+.. code-block:: console
+
+  .generated/indexes/<model_name>_cache.pkl
+
+If instead you would like to use your own embedder, you can use the provided ``Embedder`` abstract class. You need to implement two methods: ``load`` and ``encode``. The load method will load and return your embedder model. The encode method will take a list of text strings and return a list of numpy vectors. You can register your class for use with MindMeld via the ``register_embedder`` method as shown below.
+
+.. code-block:: python
+
+  from mindmeld.models import Embedder, register_embedder 
+  
+  class MyEmbedder(Embedder): 
+      def load(self, **kwargs): 
+          # return the loaded model 
+      def encode(self, text_list): 
+          # encodes each query in the list  
+
+  register_embedder('my_embedder_name', MyEmbedder) 
+
+
+In many cases, you may like to provide some parameters to your load method to initialize your model in a certain way. The ``model_settings`` dictionary in the config will be passed to the load method as kwargs, so any needed parameters can be specified there.
+
+
+.. code-block:: python
+
+  QUESTION_ANSWERER_CONFIG = { 
+      "model_type": "embedder", 
+      "model_settings": { 
+          "embedder_type": "my_embedder_name", 
+          "some_model_param": "my_model_param", 
+          "embedding_fields": {"faq": ['question', 'answer']} 
+      } 
+  } 
+
+
+Once your KB has been created, to search the knowledge base leveraging vector similarity, we will use the :meth:`get()` API as before with one small modification. We set the ``query_type`` parameter to ``embedder``, ``embedder_keyword``, or ``embedder_text``. This will automatically find results for which the embedded search fields are close in cosine similarity to the embedded query.
```

Comment:
```suggestion
Once your knowledge base has been created, to search it while leveraging vector similarity, we will use the :meth:`get()` API as before with one small modification. We set the ``query_type`` parameter to ``embedder``, ``embedder_keyword``, or ``embedder_text``. This will automatically find results for which the embedded search fields are close in cosine similarity to the embedded query.
```